lucene/core/src/java/org/apache/lucene/store/IOContext.java

```diff
@@ -90,6 +90,9 @@ private IOContext(boolean readOnce, boolean load, boolean randomAccess) {
     if (readOnce && randomAccess) {
       throw new IllegalArgumentException("cannot be both readOnce and randomAccess");
     }
+    if (load && randomAccess == false) {
+      throw new IllegalArgumentException("cannot be load but not randomAccess");
```

Comment:
This looks right to me. ++

Reply:
Trivially, maybe add something like this to the javadoc for `load`:

  `If this flag is set, then randomAccess will be true.`

Reply:
Good call, done.